src/Weft/Internal/Types.hs

```diff
@@ -119,13 +118,23 @@ instance {-# OVERLAPPING #-}
            , fmap Just $ (,) <$> arbitrary <*> arbitrary
            ] `suchThat` maybe (isJust $ isAllMaybe @args) (const True)
 
+data Empty (ts :: TypeState) =
+  Empty { dontTouch :: Magic ts Void }
```

Comment:
This would actually be better as `Magic ts ()`; then we would needed the undefineds everywhere

Reply:
and then you can do the usual deriving stuff too!